oauthproxy.go

```diff
@@ -579,16 +581,43 @@ func (p *OAuthProxy) GetRedirect(req *http.Request) (redirect string, err error)
 		redirect = req.Form.Get("rd")
 	}
 	if !p.IsValidRedirect(redirect) {
-		// Use RequestURI to preserve ?query
-		redirect = req.URL.RequestURI()
-		if strings.HasPrefix(redirect, p.ProxyPrefix) {
-			redirect = "/"
+		// Quirk: On reverse proxies that doesn't have support for
+		// "X-Auth-Request-Redirect" header or dynamic header/query string
+		// manipulation (like Traefik v1 and v2), we can try if the header
+		// X-Forwarded-Host exists or not.
+		if isForwardedRequest(req) && p.ReverseProxy {
+			redirect = p.getRedirectFromForwardHeaders(req)
```

Comment:
If we load the redirect from headers here, it doesn't go through the validation, surely we want it to still be validated?

Perhaps this needs to be moved out and be more like below
```
if redirect == "" && p.ReverseProxy && isForwardedRequest(req) {
  redirect = p.getRedirectFromForwardHeaders(req)
}
if !p.isValidRedirect(redirect) {
 // As before
}
```

Reply:
This has been done.